### PR TITLE
Make skill data sources configurable

### PR DESCRIPTION
**Pull request by NeonDaniel**
Refactor skill to simplify logic
Make API skills configurable
Handle returned icon path in idle.qml

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 import os
-import sys
 import time
 import requests
+
 from os import path
-from pathlib import Path
-from PIL import Image 
+from mycroft_bus_client import Message
+from neon_utils.log_utils import LOG
```
Comment (JarbasAl) [inline __init__.py:8]: can we use the LOG from ovos_utils here?

Comment (NeonDaniel) [reply __init__.py]: Good catch; was missing that in requirements too. Fixed.

**Comment by NeonDaniel** [comment]
As uncovered by @ChanceNCounter this PR prematurely moved to an API method for datetime. Should this PR wait for an update to the datetime skill, or implement the legacy import method? https://github.com/OpenVoiceOS/skill-ovos-homescreen/blob/adb9a611b144dada656b8a8990c308c97c02cae7/__init__.py#L62

**Comment by ChanceNCounter** [comment]
Keep this branch, maybe, and patch it later? Just my 2 cents

**Comment by NeonDaniel** [comment]
> Keep this branch, maybe, and patch it later? Just my 2 cents

I spoke too soon. I did incorporate the patch, just incorrectly. Should be fixed with the previous 2 commits.

**Pull request merged**